lib/Sema/TypeCheckDecl.cpp

```diff
@@ -3371,27 +3366,44 @@ static void checkVarBehavior(VarDecl *decl, TypeChecker &TC) {
     return;
   }
 
-  // Build a Substitution vector from the conformance.
-  auto conformanceMem =
-    TC.Context.AllocateUninitialized<ProtocolConformanceRef>(1);
-  auto selfConformance = new ((void*)conformanceMem.data())
-    ProtocolConformanceRef(conformance);
-  Substitution allInterfaceSubs[] = {
-    Substitution(behaviorInterfaceSelf, *selfConformance),
-    Substitution(decl->getInterfaceType(), valueSub.getConformances()),
-  };
-  Substitution allContextSubs[] = {
-    Substitution(behaviorSelf, *selfConformance),
-    Substitution(decl->getType(), valueSub.getConformances()),
-  };
+  // Build interface and context substitution maps.
+  auto protoSelfType = behaviorProto->getProtocolSelfType();
+  auto sig = behaviorProto->getGenericSignatureOfContext();
+  auto interfaceSubsMap =
```

Comment:
This is SubstitutionMap::getProtocolSubstitutions()

Reply:
Oh, of course it is! Thanks.